addons/analytic/models/analytic_distribution_model.py

```diff
@@ -74,9 +74,9 @@ def _get_distribution(self, vals):
 
     def _get_fields_to_check(self):
         return (
-                set(self.env['account.analytic.distribution.model']._fields)
-                - set(self.env['analytic.mixin']._fields)
-                - set(models.MAGIC_COLUMNS) - {'display_name', '__last_update'}
+            {field for field in self._fields if not field.startswith('x_')}
```

Comment:
```suggestion
            {field for field in self._fields.values() if not field.manual}
```
Slightly more robust.